src/components/button/icon-button.ts

```diff
@@ -0,0 +1,92 @@
+import { html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { IgcButtonBaseComponent } from './button-base.js';
+import { styles } from './icon-button.material.css';
+
+/**
+ * @element igc-icon-button
+ *
+ * @csspart base - The wrapping element.
+ * @csspart icon - The icon element.
+ */
+export class IgcIconButtonComponent extends IgcButtonBaseComponent {
+  static styles = [styles];
+
+  /** The name of the icon. */
+  @property()
+  name!: string;
+
+  /** The name of the icon collection. */
+  @property()
+  collection!: string;
+
+  /** Whether to flip the icon. Useful for RTL layouts. */
+  @property({ type: Boolean })
+  mirrored = false;
+
+  /** The href attribute of the icon. */
```

Comment:
From here on, the comments should probably mention `icon button` instead of just `icon`.